### DISCUSSION TITLE
Unable to run seeds in jest test

### DISCUSSION THREAD

**Discussion by anzau** [tada5hi/typeorm-extension#336] [Q&A]
Hi there, 

I am trying to set up Jest with typeorm-extension and I'm unable to seed the createdDB. I'm not using the factories currently for the seeds but when I run it on production and development environment, the seeds run fine with the same code but not with Jest. The only change I have is the `createDatabase` function but that does its job well in relation to creating a database instance with the given options. Here is the following code for your reference:

```typescript
export let typeormDataSource: DataSource;
const opt: DataSourceOptions & SeederOptions = {
	type: 'mysql',
	host: config.DB_HOST || '127.0.0.1',
	port: <number>(<unknown>config.DB_PORT) || 3306,
	username: config.DB_USER,
	password: config.DB_PASSWORD,
	database: 'test_db',
	entities: ['src/database/entities/*{.ts,.js}'],
	migrations: ['src/database/migrations/*{.ts,.js}'],
	migrationsRun: true,
	logging: true,
	synchronize: false,
	seeds: ['src/database/seeds/*/**{.ts,.js}'],
	dropSchema: true
};

beforeAll(async () => {
	await createDatabase({
		options: opt,
		ifNotExist: true
	});

	typeormDataSource = new DataSource(opt);
	
	await typeormDataSource.initialize();
	
	await runSeeders(typeormDataSource);

});
```

**Comment by tada5hi** [comment]
Hey,

that is kind of strange. Do you have the same data-source configuration when running in production/development ?
The seed glob string might be wrong depending on your file structure.

```typescript
match("src/database/seeds/abc.ts", "src/database/seeds/**/*{.ts,.js}") <--- true
match("src/database/seeds/abc.ts", "src/database/seeds/*/**{.ts,.js}") <-- false

match("src/database/seeds/subdirectory/abc.ts", "src/database/seeds/**/*{.ts,.js}") <--- true
match("src/database/seeds/subdirectory/abc.ts", "src/database/seeds/*/**{.ts,.js}") <--- true
```

**Reply by anzau** [reply]
Hey,

Yeah, the config and the path is all the same with the exception of the database name and the drop schema which is false. I tried changing the path with "../../src/database/seeds/**/*{.ts.js}" as I have it within this base test file in a folder but that didn't seem to work either. It's also strange because the migration and the entities path are similar and they seem to be working fine but not the seeds. I've also tried to log a message after that runSeerders where it is logging but the function is not throwing any errors.

**Reply by tada5hi** [reply]
@anzau could you create an example repository for reproduction?
That would be realy great.
Greetings
Peter
